src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java

```diff
@@ -269,20 +277,50 @@ private ContextMenu createFileMenu(BibEntryTableViewModel entry, List<LinkedFile
     }
 
     /**
-     * Creates a column which shows an icon instead of the textual content
+     * Creates a column for DOIs or URLs.
+     * The {@code firstField} is preferred to be shown over {@code secondField}.
      */
-    private TableColumn<BibEntryTableViewModel, String> createIconColumn(JabRefIcon icon, String firstField, String secondField) {
+    private TableColumn<BibEntryTableViewModel, String> createUrlOrDoiColumn(JabRefIcon icon, String firstField, String secondField) {
         TableColumn<BibEntryTableViewModel, String> column = new TableColumn<>();
         column.setGraphic(icon.getGraphicNode());
         column.getStyleClass().add(ICON_COLUMN);
         setExactWidth(column, GUIGlobals.WIDTH_ICON_COL);
-        column.setCellValueFactory(cellData -> EasyBind.monadic(cellData.getValue().getField(firstField)).orElse(cellData.getValue().getField(secondField)));
+        column.setCellValueFactory(cellData -> EasyBind.monadic(cellData.getValue().getField(firstField)).map(x -> firstField).orElse(EasyBind.monadic(cellData.getValue().getField(secondField)).map(x -> secondField)));
                 new ValueTableCellFactory<BibEntryTableViewModel, String>()
                         .withGraphic(cellFactory::getTableIcon)
+                        .withOnMouseClickedEvent((entry, content) -> createOpenUrlOrDoiHandler(entry, firstField, secondField))
```

Comment:
I hope a construction like `(entry, content) -> event -> openurlOrDoi(event, entry, content)` with method `openUrlOrDoi(MouseEvent event, BibentryTableViewmodel entry, String field)` should work and is a bit easier to parse.

Reply:
Done!